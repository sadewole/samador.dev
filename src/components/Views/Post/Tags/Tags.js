import React from "react"
import PropTypes from "prop-types"

const Tags = ({ tags }) => {
  return (
    <div>
      <ul className="tags">
        {tags &&
          tags.map((tag, index) => (
            <li
              className="text-white hover:border-gray-600 border-white border p-3 mr-2 text-center inline-flex items-center justify-center h-10 mb-6 shadow-lg rounded-full hover:bg-white hover:text-gray-600 cursor-pointer"
              key={index}
            >
              {tag}
            </li>
          ))}
      </ul>
    </div>
  )
}

Tags.defaultProps = {
  tags: [],
}

Tags.propTypes = {
  tags: PropTypes.array,
}

export default Tags
