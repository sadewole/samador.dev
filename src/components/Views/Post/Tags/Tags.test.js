import React from "react"
import renderer from "react-test-renderer"
import Tags from "./Tags"

describe("Tags", () => {
  const props = {
    tags: ["test"],
  }

  it("render correctly", () => {
    const tree = renderer.create(<Tags {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
