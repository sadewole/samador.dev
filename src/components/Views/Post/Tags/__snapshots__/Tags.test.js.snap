// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags render correctly 1`] = `
<div>
  <ul
    className="tags"
  >
    <li
      className="text-white hover:border-gray-600 border-white border p-3 mr-2 text-center inline-flex items-center justify-center h-10 mb-6 shadow-lg rounded-full hover:bg-white hover:text-gray-600 cursor-pointer"
    >
      test
    </li>
  </ul>
</div>
`;
